### ISSUE TITLE
Examples of API usage

### ISSUE THREAD

**Issue by Gozem** [bus1/bus1#38]
Hi,

I am doing some research around this nice new feature for a Company. It all seems very nice, but I am missing some simple "get started" examples of the API. I did use the unit-tests as examples, but that is not optimal, and lacks documentation/minor explanations.

I had to read much of the code-base to get started. Some small examples would be great and will also faster increase the user base.

**Comment by dvdhrm** [comment]
Yes, we definitely should provide this! I just ask you to wait a couple of weeks until we send the first RFC to LKML. Starting with that submission we will do proper ABI tracking and code management.

Furthermore, we have a rather big ABI change pending which will introduce mandatory 'unmanaged' nodes and thus change how the ID-management of handles works.

**Comment by breznak** [comment]
And the RFC is posted to LKML http://lkml.iu.edu/hypermail/linux/kernel/1610.3/02995.html 
Congratz guys! :+1: 
You are referring to many prior-art implementations (Google's Binder, Cap'n'P, ...): please try to get respective people from the projects to review your design (you mentioned talks with google). It'll be great if this project implements the latest & best knowledge from the others.

**Comment by cdancy** [comment]
Piggy backing off of this thread as it's _sort of_ related: are there docker/rkt images which has this module baked in possibly for testing purposes? My motivation is around a java library which will call  this module/sys-call. I've got at least a thousand use-cases for this so I'm eager to start putting something together sooner, rather than later, with hopes of this eventually being mainlined into the kernel.

**Comment by gregkh** [comment]
Docker/rkt images do not include a kernel module...

**Comment by cdancy** [comment]
@gregkh yeah you're right. Vagrant image possibly? Just something to get up and running quick is all I'm wondering.

**Comment by gregkh** [comment]
@cdancy take your favorite image today and just build the code for it and load it, it shouldn't be hard, any distro image can do this already, no need to make a special one for it.

**Comment by gregkh** [comment]
@cdancy or just do it on your desktop, it is running Linux, right?  :)

**Comment by cdancy** [comment]
@gregkh work machine I'm on now is osx or as it's now known as again: mac. 

I can do things easily enough at home or in a VM but with you guys playing I wasn't sure if you were doing something similar that I could take advantage of w/out having to go through the whole setup process and potentially hitting issue X.

Anyway ... I'll go that route. Thanks for chiming in.
